### PR TITLE
fix(dal): Add index on createdAt for Online Archive in execution, job…

### PR DESCRIPTION
**Pull request by merrcury**
…, message, and notification schemas

### What changed? Why was the change needed?

This is required for MongoDB Online Archive 

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: libs/dal/src/repositories/execution-details/execution-details.schema.ts [hunk 1]
```diff
@@ -111,6 +110,12 @@ executionDetailsSchema.index({
   _environmentId: 1,
 });
 
+/*
+ * This index was created to push entries to Online Archive
+ */
+
+executionDetailsSchema.index({ createdAt: 1 });
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/execution-details/execution-details.schema.ts]: i think we don't need the extra empty line here 😎
```suggestion
executionDetailsSchema.index({ createdAt: 1 });
```

File: libs/dal/src/repositories/job/job.schema.ts [hunk 2]
```diff
@@ -397,4 +396,10 @@ jobSchema.index(
   }
 );
 
+/*
+ * This index was created to push entries to Online Archive
+ */
+
+jobSchema.index({ createdAt: 1 });
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/job/job.schema.ts]: same here :)
```suggestion
 */
jobSchema.index({ createdAt: 1 });
```

File: libs/dal/src/repositories/message/message.schema.ts [hunk 3]
```diff
@@ -288,6 +287,12 @@ messageSchema.index({
   createdAt: 1,
 });
 
+/*
+ * This index was created to push entries to Online Archive
+ */
+
+messageSchema.index({ createdAt: 1 });
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/message/message.schema.ts]: same here :)
```suggestion
 */
messageSchema.index({ createdAt: 1 });
```

File: libs/dal/src/repositories/notification/notification.schema.ts [hunk 4]
```diff
@@ -155,6 +154,12 @@ notificationSchema.index({
   createdAt: -1,
 });
 
+/*
+ * This index was created to push entries to Online Archive
+ */
+
+notificationSchema.index({ createdAt: 1 });
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/notification/notification.schema.ts]: same here 
```suggestion
 */
notificationSchema.index({ createdAt: 1 });
```

**Review by djabarovgeorge — APPROVED** [review]

**Comment by djabarovgeorge** [comment]
Looks good :) 
We will need to make sure to add those to the mongo instance in a rolling manner

File: libs/dal/src/repositories/notification/notification.schema.ts (hunk 4 above)
Comment (merrcury) [reply libs/dal/src/repositories/notification/notification.schema.ts]: Done

File: libs/dal/src/repositories/message/message.schema.ts (hunk 3 above)
Comment (merrcury) [reply libs/dal/src/repositories/message/message.schema.ts]: Done

File: libs/dal/src/repositories/job/job.schema.ts (hunk 2 above)
Comment (merrcury) [reply libs/dal/src/repositories/job/job.schema.ts]: Done

File: libs/dal/src/repositories/execution-details/execution-details.schema.ts (hunk 1 above)
Comment (merrcury) [reply libs/dal/src/repositories/execution-details/execution-details.schema.ts]: Done

**Pull request merged**